src/wallet/db.cpp

```diff
@@ -56,6 +51,11 @@ CCriticalSection cs_db;
 std::map<std::string, BerkeleyEnvironment> g_dbenvs GUARDED_BY(cs_db); //!< Map from directory name to open db environment.
 } // namespace
 
+bool BerkeleyFileid::operator==(const BerkeleyFileid& b) const
+{
+    return memcmp(this, &b, sizeof(BerkeleyFileid)) == 0;
```

Comment:
It might be safer if memcmp compared just the value field rather than the whole object, since it seems possible the compiler could pad the objects. Maybe:

```
return memcmp(this->field, b.field, sizeof(this->field)) == 0;
```